byob.js

```diff
@@ -292,11 +292,17 @@ CustomBlockDefinition.prototype.inputOptionsOfIdx = function (idx) {
 };
 
 CustomBlockDefinition.prototype.dropDownMenuOf = function (inputName) {
-    var dict = {};
+    var dict = {},
+	    gensym = 1;
     if (this.declarations[inputName] && this.declarations[inputName][2]) {
         this.declarations[inputName][2].split('\n').forEach(function (line) {
-            var pair = line.split('=');
-            dict[pair[0]] = isNil(pair[1]) ? pair[0] : pair[1];
+            var pair = line.split('='),
+			    unique = pair[0];
+			if (unique == '~' || unique == '}') {
+				unique = unique.concat(gensym.toString());
```

Comment:
Unless @jmoenig disagrees with this, I think this line can be simplified `unique += gensym` does exactly the same thing, but it might be better to be a little more explicit. At the very least `.toString()` is unnecessary.
```js
 > a = '~'
'~'
> a += 1
'~1'
> 
```
(For all the quirks of JS coercion, a string + an int is a string, is a consistent rule.)